hail/src/main/scala/is/hail/linalg/BlockMatrix.scala

```diff
@@ -249,22 +253,25 @@ object BlockMatrix {
                 val n = stopColOffsetInBlock - startColOffsetInBlock
 
                 inPerBlockCol(blockCol - startBlockCol).readDoubles(data, 0, n)
+                
+                if (writeBytes) {
+                  os.asInstanceOf[DoubleOutputBuffer].writeDoubles(data, 0, n)
```

Comment:
The "right way" to do this would be a lambda or the equivalent subclass-based abstraction. For example:

```
val writeData = if (writeBytes) {
```

Reply:
This ensures that the conditional check is done in one place in the code, which hopefully makes it more clear to the reader that these casts are safe.

Reply:
I think we should do this.